// react-native-gesture-handler/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Directions", {
  enumerable: true,
  get: function () {
    return _Directions.Directions;
  }
});
Object.defineProperty(exports, "State", {
  enumerable: true,
  get: function () {
    return _State.State;
  }
});
Object.defineProperty(exports, "gestureHandlerRootHOC", {
  enumerable: true,
  get: function () {
    return _gestureHandlerRootHOC.default;
  }
});
Object.defineProperty(exports, "GestureHandlerRootView", {
  enumerable: true,
  get: function () {
    return _GestureHandlerRootView.default;
  }
});
Object.defineProperty(exports, "GestureEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.GestureEvent;
  }
});
Object.defineProperty(exports, "HandlerStateChangeEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.HandlerStateChangeEvent;
  }
});
Object.defineProperty(exports, "GestureEventPayload", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.GestureEventPayload;
  }
});
Object.defineProperty(exports, "HandlerStateChangeEventPayload", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.HandlerStateChangeEventPayload;
  }
});
Object.defineProperty(exports, "TapGestureHandlerEventPayload", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.TapGestureHandlerEventPayload;
  }
});
Object.defineProperty(exports, "ForceTouchGestureHandlerEventPayload", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.ForceTouchGestureHandlerEventPayload;
  }
});
Object.defineProperty(exports, "LongPressGestureHandlerEventPayload", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.LongPressGestureHandlerEventPayload;
  }
});
Object.defineProperty(exports, "PanGestureHandlerEventPayload", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.PanGestureHandlerEventPayload;
  }
});
Object.defineProperty(exports, "PinchGestureHandlerEventPayload", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.PinchGestureHandlerEventPayload;
  }
});
Object.defineProperty(exports, "RotationGestureHandlerEventPayload", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.RotationGestureHandlerEventPayload;
  }
});
Object.defineProperty(exports, "FlingGestureHandlerEventPayload", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.FlingGestureHandlerEventPayload;
  }
});
Object.defineProperty(exports, "TapGestureHandlerProps", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.TapGestureHandlerProps;
  }
});
Object.defineProperty(exports, "ForceTouchGestureHandlerProps", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.ForceTouchGestureHandlerProps;
  }
});
Object.defineProperty(exports, "LongPressGestureHandlerProps", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.LongPressGestureHandlerProps;
  }
});
Object.defineProperty(exports, "PanGestureHandlerProps", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.PanGestureHandlerProps;
  }
});
Object.defineProperty(exports, "PinchGestureHandlerProps", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.PinchGestureHandlerProps;
  }
});
Object.defineProperty(exports, "RotationGestureHandlerProps", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.RotationGestureHandlerProps;
  }
});
Object.defineProperty(exports, "FlingGestureHandlerProps", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.FlingGestureHandlerProps;
  }
});
Object.defineProperty(exports, "TapGestureHandler", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.TapGestureHandler;
  }
});
Object.defineProperty(exports, "ForceTouchGestureHandler", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.ForceTouchGestureHandler;
  }
});
Object.defineProperty(exports, "LongPressGestureHandler", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.LongPressGestureHandler;
  }
});
Object.defineProperty(exports, "PanGestureHandler", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.PanGestureHandler;
  }
});
Object.defineProperty(exports, "PinchGestureHandler", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.PinchGestureHandler;
  }
});
Object.defineProperty(exports, "RotationGestureHandler", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.RotationGestureHandler;
  }
});
Object.defineProperty(exports, "FlingGestureHandler", {
  enumerable: true,
  get: function () {
    return _gestureHandlers.FlingGestureHandler;
  }
});
Object.defineProperty(exports, "createNativeWrapper", {
  enumerable: true,
  get: function () {
    return _createNativeWrapper.default;
  }
});
Object.defineProperty(exports, "NativeViewGestureHandler", {
  enumerable: true,
  get: function () {
    return _NativeViewGestureHandler.NativeViewGestureHandler;
  }
});
Object.defineProperty(exports, "NativeViewGestureHandlerPayload", {
  enumerable: true,
  get: function () {
    return _NativeViewGestureHandler.NativeViewGestureHandlerPayload;
  }
});
Object.defineProperty(exports, "NativeViewGestureHandlerProps", {
  enumerable: true,
  get: function () {
    return _NativeViewGestureHandler.NativeViewGestureHandlerProps;
  }
});
Object.defineProperty(exports, "RawButtonProps", {
  enumerable: true,
  get: function () {
    return _GestureButtons.RawButtonProps;
  }
});
Object.defineProperty(exports, "BaseButtonProps", {
  enumerable: true,
  get: function () {
    return _GestureButtons.BaseButtonProps;
  }
});
Object.defineProperty(exports, "RectButtonProps", {
  enumerable: true,
  get: function () {
    return _GestureButtons.RectButtonProps;
  }
});
Object.defineProperty(exports, "BorderlessButtonProps", {
  enumerable: true,
  get: function () {
    return _GestureButtons.BorderlessButtonProps;
  }
});
Object.defineProperty(exports, "RawButton", {
  enumerable: true,
  get: function () {
    return _GestureButtons.RawButton;
  }
});
Object.defineProperty(exports, "BaseButton", {
  enumerable: true,
  get: function () {
    return _GestureButtons.BaseButton;
  }
});
Object.defineProperty(exports, "RectButton", {
  enumerable: true,
  get: function () {
    return _GestureButtons.RectButton;
  }
});
Object.defineProperty(exports, "BorderlessButton", {
  enumerable: true,
  get: function () {
    return _GestureButtons.BorderlessButton;
  }
});
Object.defineProperty(exports, "TouchableHighlight", {
  enumerable: true,
  get: function () {
    return _touchables.TouchableHighlight;
  }
});
Object.defineProperty(exports, "TouchableNativeFeedback", {
  enumerable: true,
  get: function () {
    return _touchables.TouchableNativeFeedback;
  }
});
Object.defineProperty(exports, "TouchableOpacity", {
  enumerable: true,
  get: function () {
    return _touchables.TouchableOpacity;
  }
});
Object.defineProperty(exports, "TouchableWithoutFeedback", {
  enumerable: true,
  get: function () {
    return _touchables.TouchableWithoutFeedback;
  }
});
Object.defineProperty(exports, "ScrollView", {
  enumerable: true,
  get: function () {
    return _GestureComponents.ScrollView;
  }
});
Object.defineProperty(exports, "Switch", {
  enumerable: true,
  get: function () {
    return _GestureComponents.Switch;
  }
});
Object.defineProperty(exports, "TextInput", {
  enumerable: true,
  get: function () {
    return _GestureComponents.TextInput;
  }
});
Object.defineProperty(exports, "DrawerLayoutAndroid", {
  enumerable: true,
  get: function () {
    return _GestureComponents.DrawerLayoutAndroid;
  }
});
Object.defineProperty(exports, "FlatList", {
  enumerable: true,
  get: function () {
    return _GestureComponents.FlatList;
  }
});
Object.defineProperty(exports, "GestureHandlerGestureEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.GestureHandlerGestureEvent;
  }
});
Object.defineProperty(exports, "GestureHandlerStateChangeEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.GestureHandlerStateChangeEvent;
  }
});
Object.defineProperty(exports, "GestureHandlerGestureEventNativeEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.GestureHandlerGestureEventNativeEvent;
  }
});
Object.defineProperty(exports, "GestureHandlerStateChangeNativeEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.GestureHandlerStateChangeNativeEvent;
  }
});
Object.defineProperty(exports, "NativeViewGestureHandlerGestureEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.NativeViewGestureHandlerGestureEvent;
  }
});
Object.defineProperty(exports, "NativeViewGestureHandlerStateChangeEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.NativeViewGestureHandlerStateChangeEvent;
  }
});
Object.defineProperty(exports, "TapGestureHandlerGestureEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.TapGestureHandlerGestureEvent;
  }
});
Object.defineProperty(exports, "TapGestureHandlerStateChangeEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.TapGestureHandlerStateChangeEvent;
  }
});
Object.defineProperty(exports, "ForceTouchGestureHandlerGestureEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.ForceTouchGestureHandlerGestureEvent;
  }
});
Object.defineProperty(exports, "ForceTouchGestureHandlerStateChangeEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.ForceTouchGestureHandlerStateChangeEvent;
  }
});
Object.defineProperty(exports, "LongPressGestureHandlerGestureEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.LongPressGestureHandlerGestureEvent;
  }
});
Object.defineProperty(exports, "LongPressGestureHandlerStateChangeEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.LongPressGestureHandlerStateChangeEvent;
  }
});
Object.defineProperty(exports, "PanGestureHandlerGestureEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.PanGestureHandlerGestureEvent;
  }
});
Object.defineProperty(exports, "PanGestureHandlerStateChangeEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.PanGestureHandlerStateChangeEvent;
  }
});
Object.defineProperty(exports, "PinchGestureHandlerGestureEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.PinchGestureHandlerGestureEvent;
  }
});
Object.defineProperty(exports, "PinchGestureHandlerStateChangeEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.PinchGestureHandlerStateChangeEvent;
  }
});
Object.defineProperty(exports, "RotationGestureHandlerGestureEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.RotationGestureHandlerGestureEvent;
  }
});
Object.defineProperty(exports, "RotationGestureHandlerStateChangeEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.RotationGestureHandlerStateChangeEvent;
  }
});
Object.defineProperty(exports, "FlingGestureHandlerGestureEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.FlingGestureHandlerGestureEvent;
  }
});
Object.defineProperty(exports, "FlingGestureHandlerStateChangeEvent", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.FlingGestureHandlerStateChangeEvent;
  }
});
Object.defineProperty(exports, "NativeViewGestureHandlerProperties", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.NativeViewGestureHandlerProperties;
  }
});
Object.defineProperty(exports, "TapGestureHandlerProperties", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.TapGestureHandlerProperties;
  }
});
Object.defineProperty(exports, "LongPressGestureHandlerProperties", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.LongPressGestureHandlerProperties;
  }
});
Object.defineProperty(exports, "PanGestureHandlerProperties", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.PanGestureHandlerProperties;
  }
});
Object.defineProperty(exports, "PinchGestureHandlerProperties", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.PinchGestureHandlerProperties;
  }
});
Object.defineProperty(exports, "RotationGestureHandlerProperties", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.RotationGestureHandlerProperties;
  }
});
Object.defineProperty(exports, "FlingGestureHandlerProperties", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.FlingGestureHandlerProperties;
  }
});
Object.defineProperty(exports, "ForceTouchGestureHandlerProperties", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.ForceTouchGestureHandlerProperties;
  }
});
Object.defineProperty(exports, "RawButtonProperties", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.RawButtonProperties;
  }
});
Object.defineProperty(exports, "BaseButtonProperties", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.BaseButtonProperties;
  }
});
Object.defineProperty(exports, "RectButtonProperties", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.RectButtonProperties;
  }
});
Object.defineProperty(exports, "BorderlessButtonProperties", {
  enumerable: true,
  get: function () {
    return _gestureHandlerTypesCompat.BorderlessButtonProperties;
  }
});
Object.defineProperty(exports, "Swipeable", {
  enumerable: true,
  get: function () {
    return _Swipeable.default;
  }
});
Object.defineProperty(exports, "DrawerLayout", {
  enumerable: true,
  get: function () {
    return _DrawerLayout.default;
  }
});
Object.defineProperty(exports, "DrawerLayoutProps", {
  enumerable: true,
  get: function () {
    return _DrawerLayout.DrawerLayoutProps;
  }
});
Object.defineProperty(exports, "DrawerPosition", {
  enumerable: true,
  get: function () {
    return _DrawerLayout.DrawerPosition;
  }
});
Object.defineProperty(exports, "DrawerState", {
  enumerable: true,
  get: function () {
    return _DrawerLayout.DrawerState;
  }
});
Object.defineProperty(exports, "DrawerType", {
  enumerable: true,
  get: function () {
    return _DrawerLayout.DrawerType;
  }
});
Object.defineProperty(exports, "DrawerLockMode", {
  enumerable: true,
  get: function () {
    return _DrawerLayout.DrawerLockMode;
  }
});
Object.defineProperty(exports, "DrawerKeyboardDismissMode", {
  enumerable: true,
  get: function () {
    return _DrawerLayout.DrawerKeyboardDismissMode;
  }
});

var _Directions = require("./Directions");

var _State = require("./State");

var _gestureHandlerRootHOC = _interopRequireDefault(require("./gestureHandlerRootHOC"));

var _GestureHandlerRootView = _interopRequireDefault(require("./GestureHandlerRootView"));

var _gestureHandlers = require("./handlers/gestureHandlers");

var _createNativeWrapper = _interopRequireDefault(require("./handlers/createNativeWrapper"));

var _NativeViewGestureHandler = require("./handlers/NativeViewGestureHandler");

var _GestureButtons = require("./components/GestureButtons");

var _touchables = require("./components/touchables");

var _GestureComponents = require("./components/GestureComponents");

var _gestureHandlerTypesCompat = require("./handlers/gestureHandlerTypesCompat");

var _Swipeable = _interopRequireDefault(require("./components/Swipeable"));

var _DrawerLayout = _interopRequireWildcard(require("./components/DrawerLayout"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
